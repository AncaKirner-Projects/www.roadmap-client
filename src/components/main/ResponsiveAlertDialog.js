import React from 'react';
import Button from '@material-ui/core/Button';
import Dialog from '@material-ui/core/Dialog';
import DialogActions from '@material-ui/core/DialogActions';
import DialogContent from '@material-ui/core/DialogContent';
import DialogContentText from '@material-ui/core/DialogContentText';
import DialogTitle from '@material-ui/core/DialogTitle';

const ResponsiveDialog = (props) => {
  const { fullScreen } = props;

  return (
    <Dialog
      fullScreen={fullScreen}
      open={props.open}
      onClose={props.handleClose}
      aria-labelledby="responsive-dialog-title"
    >
      <DialogTitle id="responsive-dialog-title">Confirmation</DialogTitle>
      <DialogContent>
        <DialogContentText>
          Are you sure you want to delete the product?
        </DialogContentText>
      </DialogContent>
      <DialogActions>
        <Button onClick={props.onHandleClose} color="primary">
          No
        </Button>
        <Button onClick={props.onHandleConfirmClose} color="primary" autoFocus>
          Yes
        </Button>
      </DialogActions>
    </Dialog>
  );
};

export default ResponsiveDialog;
