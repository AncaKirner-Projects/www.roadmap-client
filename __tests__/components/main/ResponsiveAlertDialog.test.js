import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import ResponsiveAlertDialog from '../../../src/components/main/ResponsiveAlertDialog';

describe('<ResponsiveAlertDialog />', () => {
  const props = {
    fullScreen: false,
    open: true,
    handleClose: jest.fn(),
    handleConfirmClose: jest.fn()
  };

  it('renders an element', () => {
    const wrapper = shallow(<ResponsiveAlertDialog {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
