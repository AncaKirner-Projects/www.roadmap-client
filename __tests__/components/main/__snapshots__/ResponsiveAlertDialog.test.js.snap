// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ResponsiveAlertDialog /> renders an element 1`] = `
<WithStyles(Dialog)
  aria-labelledby="responsive-dialog-title"
  fullScreen={false}
  onClose={[MockFunction]}
  open={true}
>
  <WithStyles(DialogTitle)
    id="responsive-dialog-title"
  >
    Confirmation
  </WithStyles(DialogTitle)>
  <WithStyles(DialogContent)>
    <WithStyles(DialogContentText)>
      Are you sure you want to delete the product?
    </WithStyles(DialogContentText)>
  </WithStyles(DialogContent)>
  <WithStyles(DialogActions)>
    <WithStyles(Button)
      color="primary"
      onClick={[MockFunction]}
    >
      No
    </WithStyles(Button)>
    <WithStyles(Button)
      autoFocus={true}
      color="primary"
      onClick={[MockFunction]}
    >
      Yes
    </WithStyles(Button)>
  </WithStyles(DialogActions)>
</WithStyles(Dialog)>
`;
